DevREADME.md

```diff
@@ -99,4 +99,132 @@ You may encounter the failure of npm install, the error message would say:
 Error installing vscode.d.ts: Error: read ECONNRESET
 If you encounter this problem, maybe your NODE is too old, in version v10.13.0 it 
 can work well.
-or run "npm cache clean -force"
\ No newline at end of file
+or run "npm cache clean -force"
+
+---
+
+## Style Guide
+Below are some general code guidelines that should be followed when adding or updating the code base. It is the expectation that participating members actively apply this style guide to legacy code during maintenance so we can eventually have a noticeably consistent developer experience.
+### Names
+1. Use `PascalCase` for type/interface names, enums, setters, getters and global singletons
+2. Use `camelCase` for function, argument and local variable names.
+3. Use `UPPER_CASE` for global constants.
+4. Use `_` as a prefix for private variables
+5. Use `I` as a prefix for interface names that define function declarations.
+5. Use whole words in names where possible.
+
+### Types
+1. Do not export types/functions unless you need to share them across multiple components.
+2. Within a file, type definitions should come first.
+
+### Comments
+1. Use JSDoc-style comments for functions, interfaces, enums, and classes.
+2. Limited to using an informal description and @type/@param/@return when applicable
+
+### General
+1. Use arrow functions over anonymous function expressions.
+2. Always surround loop and conditional bodies with curly braces.
+3. Open curly braces always go on the same line as whatever necessitates them.
+4. `else` goes on the same line as the closing curly brace.
+5. Use two (2) spaces for indentation.
+6. The `export` keyword should be on its own line.
+7. TypeScript errors/warnings need to be resolved.
+8. Function should always provide type declarations for its return and argument values
+9. Function bodies (including getters/setters) must appear on separate lines from their signature declarations regardless of their simplicity. Empty constructors that exist for clarity are acceptable on a single line
+10. Use a maximum of 140 chars per lines; including leading whitespace
```

Comment:
The original minimalist example value was 80 except for function declarations. I grabbed the value from the in-depth version because 80 was a bit to short.; IE, roughly half the width of a 1280x1024 monitor at what I believe is the default text size.